### PR TITLE
feat(Media): add fullscreen video

### PR DESCRIPTION
**Pull request by aeksandla**

### CODE REVIEW

File: src/components/Media/Image/Image.tsx [hunk 1]
```diff
@@ -58,22 +68,18 @@ const Image = (props: ImageAllProps) => {
...
-                    />
-                );
-            })}
-        </SliderBlock>
-    );
+    const fullscreenImage = (item: ImageProps) => {
+        const itemData = getMediaImage(item);
+
+        return (
+            <FullScreenImage
```
Comment (yuberdysheva) [inline src/components/Media/Image/Image.tsx:75]: `FullscreenImage`

File: src/components/Media/Image/Image.tsx [hunk 2]
```diff
@@ -28,7 +29,16 @@ interface InnerImageProps {
...
-    const {parallax, image, height, imageClassName, isBackground, hasVideoFallback, video} = props;
+    const {
+        parallax,
+        image,
+        height,
+        imageClassName,
+        isBackground,
+        hasVideoFallback,
+        video,
+        fullScreen,
```
Comment (yuberdysheva) [inline src/components/Media/Image/Image.tsx:40]: fullscreen)

File: src/components/Media/Media.tsx [hunk 3]
```diff
@@ -58,27 +59,32 @@ export const Media = (props: MediaAllProps) => {
                     isBackground={isBackground}
                     video={video}
                     hasVideoFallback={hasVideoFallback}
+                    fullScreen={fullScreen}
```
Comment (yuberdysheva) [inline src/components/Media/Media.tsx:62]: fullscreen

**Review by yuberdysheva — CHANGES_REQUESTED** [review]

**Review by yuberdysheva — APPROVED** [review]

File: src/components/FullscreenMedia/FullscreenMedia.tsx [hunk 4]
```diff
@@ -7,15 +7,24 @@ import {FullScreen, PreviewClose} from '../../icons';
 import {block} from '../../utils';
 import {MediaAllProps} from '../Media/Media';
 
-import './FullScreenMedia.scss';
+import './FullscreenMedia.scss';
 
 export type ChildMediaRenderProps = Pick<
     MediaAllProps,
-    'fullScreen' | 'imageClassName' | 'videoClassName' | 'youtubeClassName' | 'className'
+    | 'fullScreen'
```
Comment (gorgeousvlad) [inline src/components/FullscreenMedia/FullscreenMedia.tsx:14]: why should we use both fullScreen and fullscreen?

File: src/components/FullscreenMedia/FullscreenMedia.tsx [hunk 5]
```diff
@@ -25,7 +34,11 @@ const CLOSE_ICON_SIZE = 30;
 
 const getMediaClass = (type: string) => b('modal-media', {type});
 
-const FullScreenMedia = ({children, showFullScreenIcon = true}: FullScreenMediaProps) => {
+const FullscreenMedia = ({
+    children,
+    showFullScreenIcon = true,
```
Comment (gorgeousvlad) [inline src/components/FullscreenMedia/FullscreenMedia.tsx:39]: lets move it separately and add comment like `TODO: ...`
```
const getShowFullScreenIcon = ({showFullScreenIcon = true, showFullscreenIcon = true}) => showFullScreenIcon && showFullscreenIcon
```

File: src/components/Media/Image/Image.tsx [hunk 6]
```diff
@@ -58,22 +68,18 @@ const Image = (props: ImageAllProps) => {
...
-                        key={itemData.alt}
-                        {...itemData}
-                        imageClassName={imageClass}
-                        imageStyle={{height}}
-                    />
-                );
-            })}
-        </SliderBlock>
-    );
+    const fullscreenImage = (item: ImageProps) => {
```
Comment (gorgeousvlad) [inline src/components/Media/Image/Image.tsx:71]: renderFullscreenImage/getFullscreenImage

File: src/components/Media/Media.tsx [hunk 7]
```diff
@@ -29,6 +30,7 @@ export const Media = (props: MediaAllProps) => {
         parallax = false,
         metrika,
         fullScreen,
+        fullscreen,
```
Comment (gorgeousvlad) [inline src/components/Media/Media.tsx:33]: lets comment all this fullScreen and fullscreen cases to find and fix them later

or create utility function which would manage with both flags

File: src/schema/validators/common.ts [hunk 8]
```diff
@@ -512,6 +512,12 @@ export const MediaProps = {
         type: 'string',
     },
     dataLens: DataLensProps,
+    fullScreen: {
```
Comment (gorgeousvlad) [inline src/schema/validators/common.ts:515]: there was no that field in validators at all?

**Review by gorgeousvlad — CHANGES_REQUESTED** [review]

File: src/stories/COMMON_TYPES.md [hunk 9]
```diff
@@ -55,16 +55,17 @@
 
 ---
 
-# `Media`— Image/video/DataLens
+# `Media`— Image/video/DataLens/Youtube
 
 - `color?: string` — Background color
-- [`image?: ImageObjectProps | ImageObjectProps[] | ImageDeviceProps` — Background image](#ImageObjectProps)
+- [`image?: ImageObjectProps | ImageObjectProps[] | ImageDeviceProps` — Background image](#ImageObjectProps). If you pass an array, the full-screen mode is automatically turned on.
```
[the remark this answers is not in the document]
Comment (aeksandla) [reply src/stories/COMMON_TYPES.md:61]: so it was originally in this component, now I have added this moment to the doc

File: src/schema/validators/common.ts (hunk 8 above)
Comment (aeksandla) [reply src/schema/validators/common.ts:515]: yes

**Pull request merged**